network/src/transport.rs

```diff
@@ -524,405 +524,3 @@ where
         Ok((listener, listen_addr))
     }
 }
-
```

Comment:
1) This is an anti-pattern, I think the only time to move tests to their own file is when they are truly integration tests or there is some sort of common testing scheme across multiple implementations
2) This library really needs to be refactored. Transports shouldn't be this complicated.

Reply:
> This is an anti-pattern, I think the only time to move tests to their own file is when they are truly integration tests or there is some sort of common testing scheme across multiple implementations

That's definitely not a rule we follow in the codebase, and from what I've seen it's highly dependent on the crate. For example crypto moves ALL tests into test files. Having said that, I think it's enough to follow common sense on when to do this: when tests grow large, and your file is already large, it's a good idea to move tests out. It's all about readability.

Reply:
I've made this mistake before about moving to this anti-pattern.

Let's *not* move the tests because this causes a couple things:
1. I can't tell if you changed the tests
2. Later it shows you've made all the tests, and you lose all history on the changes (at least by doing git blame)

Reply:
1. I did not change the tests
2. that sounds a bit too much, this means you can't refactor anything or touch any code that was written by someone else?

Reply:
Ah, I finally figured out how to view by commit.  Yeah, now I see there were no changes.

There's an overarching discussion about what our future test breakdown should be.  I've had people argue with me when I separate the files, put it at the end of a file, and it doesn't even match our coding guidelines, which is funny this is the first one that actually matches it!

Reply:
@davidiw This is now sadly the one of the few files that now actually follows the coding guidelines https://developers.libra.org/docs/community/coding-guidelines#testing

Reply:
I know Brandon and josh talked about that, imo we should ere on the side of readability: whenever your file gets too large, and you have many tests  -> move your tests. It's all about readability for people who review code on ramp up on the codebase.

Reply:
I'm with you on that.  We've got some really long files in the codebase